### ISSUE TITLE
C++ Build Error With Generated Files

### ISSUE THREAD

**Issue by embeddetech** [protocolbuffers/protobuf#12827]
I am getting an error compiling my *.pb.cc file, where the #include "google/protobuf/port_def.inc" file is generating an error saying that the absl library is too old.

The steps I am taking:

1. git clone https://github.com/protocolbuffers/protobuf.git
2. git submodule update --init --recursive 
3. Used CMake Gui to configure the project for Visual Studio 2019, with ABSL_PROPAGATE_CXX_STD=TRUE and protobuf_WITH_ZLIB=FALSE (otherwise the generation throws errors)
4. Generated the project.
5. Built the project with VS2019
6. Used protoc.exe to generate my *.pb.cc and *.pb.h files
7. Added the *pb.cc and *.pb.h files to my project
8. Added protobuf/src to my additional includes directories
9. Built the project

I get the following errors:

In file included from ../../../[redacted]/Drivers/protobuf/gwacs.pb.h:39,
                 from ../../../[redacted]/Drivers/protobuf/[redacted].cpp:2:
../../../CMakeDependencies/protobuf/src/google/protobuf/port_def.inc:215:35: error: static assertion failed: Protobuf only supports Abseil version 20230125.3 and newer.
  215 |   (ABSL_LTS_RELEASE_VERSION > (x) || \
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~
  216 |    (ABSL_LTS_RELEASE_VERSION == (x) && ABSL_LTS_RELEASE_PATCH_LEVEL >= (y)))
      |    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../CMakeDependencies/protobuf/src/google/protobuf/port_def.inc:224:15: note: in expansion of macro ‘PROTOBUF_ABSL_MIN’
  224 | static_assert(PROTOBUF_ABSL_MIN(20230125, 3),
      |               ^~~~~~~~~~~~~~~~~

I just cloned the repository tonight, built protoc.exe, generated the .cc/.h files, and referenced the same repository source code, so everything is using the same protobuf snapshot. I didn't get this error with a version of protobuf a few weeks ago.

**Comment by acozzette** [comment]
Are you sure that your build is getting Abseil from the Git submodule? If it's getting it from somewhere else then that could explain why the version appears to be too old. We just added this version check on Monday, so that could be the immediate reason it stopped working.

**Comment by embeddetech** [comment]
Hello,
Yes, I believe so. I am following the instructions here:

And running this:

git clone https://github.com/protocolbuffers/protobuf.git
cd protobuf
git submodule update --init --recursive

Is it possible there's an issue with the abseil version check, or the dependency on abseil?

**Comment by guccang** [comment]
why protobuf is more and more difficult to use??

**Comment by MichaelGu718** [comment]
Hi, I have the same problem after builded protocol buffer following the instructions on the Readme, and installed a libre2 from github. Did you managed to solve it?

> Hello, Yes, I believe so. I am following the instructions here:
> 
> https://github.com/protocolbuffers/protobuf/blob/main/src/README.md
> 
> And running this:
> 
> git clone https://github.com/protocolbuffers/protobuf.git cd protobuf git submodule update --init --recursive
> 
> Is it possible there's an issue with the abseil version check, or the dependency on abseil?

**Comment by flavienbwk** [comment]
Got the same problem as well with MSVC 2022. Tried with :

```
set(protobuf_BUILD_SHARED_LIBS YES)
set(gRPC_ABSL_PROVIDER package)

find_package(absl REQUIRED)
target_link_libraries(myprogram PUBLIC absl::strings)

# Check if the version is found
if(absl_FOUND)
    message(STATUS "Found Abseil: ${absl_VERSION}")
else()
    message(STATUS "Abseil not found")
endif()
```

Although having :

```
[cmake] -- Found Abseil: 20230802
```

I ultimately get :

```
protobuf\src\google\protobuf\stubs\common.h:20: error : C1083: impossible to open include: 'absl/strings/string_view.h': No such file or directory
# or
protobuf/src/google/protobuf/port_def.inc:215:35: error: static assertion failed: Protobuf only supports Abseil version 20230125.3 and newer.
```

What's wrong with protobuf ?

**Comment by Sectran** [comment]
same problem
port_def.inc:173:1: error: static_assert failed due to requirement '20211102 > (20230125) || (20211102 == (20230125) && 0 >= (3))' "Protobuf only supports Abseil version 20230125.3 and newer."
static_assert(PROTOBUF_ABSL_MIN(20230125, 3)
Ironically, I am using the latest version of Abseil,version:20240116

**Comment by BinaryRyan** [comment]
port_def.inc:173:1: error: static_assert failed due to requirement '20211102 > (20230125) || (20211102 == (20230125) && 0 >= (3))' "Protobuf only supports Abseil version 20230125.3 and newer."
static_assert(PROTOBUF_ABSL_MIN(20230125, 3)
Ironically, I am using the latest version of Abseil,version:20240116

**Comment by parteimeister** [comment]
It's funny that the problem is still not solved in version v29.2 (though for some reason it only occurs in Windows). :-)
